File: apps/s-2-admin/src/api/workflowStep/WorkflowStepUpdateInput.ts
import { WorkflowWhereUniqueInput } from "../workflow/WorkflowWhereUniqueInput";
import { WorkflowReminderUpdateManyWithoutWorkflowStepsInput } from "./WorkflowReminderUpdateManyWithoutWorkflowStepsInput";

export type WorkflowStepUpdateInput = {
  stepNumber?: number;
  action?: "EMAIL_HOST" | "EMAIL_ATTENDEE" | "SMS_ATTENDEE" | "SMS_NUMBER";
  workflow?: WorkflowWhereUniqueInput;
  sendTo?: string | null;
  reminderBody?: string | null;
  emailSubject?: string | null;
  template?: "REMINDER" | "CUSTOM";
  workflowReminders?: WorkflowReminderUpdateManyWithoutWorkflowStepsInput;
};

File: apps/cactus-demo-dotnet/src/APIs/WorkflowStep/Dtos/WorkflowStepUpdateInput.cs
using CactusDemoDotnet.Core.Enums;

namespace CactusDemoDotnet.APIs.Dtos;

public class WorkflowStepUpdateInput
{
    public int? Id { get; set; }

    public int? StepNumber { get; set; }

    public ActionEnum? Action { get; set; }

    public int? Workflow { get; set; }

    public string? SendTo { get; set; }

    public string? ReminderBody { get; set; }

    public string? EmailSubject { get; set; }

    public TemplateEnum? Template { get; set; }

    public List<int>? WorkflowReminders { get; set; }
}
